{
    "id": "SOLR-6868",
    "title": "Investigate failing CollectionsAPIDistributedZkTest",
    "details": {
        "components": [
            "Tests"
        ],
        "type": "Bug",
        "labels": "",
        "fix_versions": [
            "5.0",
            "6.0"
        ],
        "affect_versions": "None",
        "status": "Resolved",
        "resolution": "Fixed",
        "priority": "Major"
    },
    "description": "CollectionsAPIDistributedZkTest has been failing (on trunk and 5x) off late.\nRunning the following \nant test  -Dtestcase=CollectionsAPIDistributedZkTest -Dtests.method=testDistribSearch -Dtests.seed=8219942E65A94EC1 -Dtests.multiplier=3 -Dtests.slow=true -Dtests.locale=nl -Dtests.timezone=America/Halifax -Dtests.asserts=false -Dtests.file.encoding=ISO-8859-1\n\ncauses the following errors/exceptions:\n\n[junit4]    > Throwable #1: org.junit.ComparisonFailure: expected:<...9942E65A94EC1-002/te[mpDir-003/solrj_test_core_props_shard1_replica1]> but was:<...9942E65A94EC1-002/te[stPropertyParamsForCreate-001/instanceDir-fmchl]>\n   [junit4]    > \tat __randomizedtesting.SeedInfo.seed([8219942E65A94EC1:3FF1A3612F62EFD]:0)\n   [junit4]    > \tat org.apache.solr.cloud.CollectionsAPIDistributedZkTest.checkInstanceDirs(CollectionsAPIDistributedZkTest.java:1154)\n   [junit4]    > \tat org.apache.solr.cloud.CollectionsAPIDistributedZkTest.testCollectionsAPI(CollectionsAPIDistributedZkTest.java:901)\n   [junit4]    > \tat org.apache.solr.cloud.CollectionsAPIDistributedZkTest.doTest(CollectionsAPIDistributedZkTest.java:205)\n   [junit4]    > \tat org.apache.solr.BaseDistributedSearchTestCase.testDistribSearch(BaseDistributedSearchTestCase.java:869)\n   [junit4]    > \tat java.lang.Thread.run(Thread.java:745)\n\n\n\nand running it without that seed (and may be with the seed too) often leads to :\n\noasu.SolrIndexWriter.rollback ERROR Exception rolling back IndexWriter org.apache.lucene.store.AlreadyClosedException: refusing to delete any files: this IndexWriter hit an unrecoverable exception",
    "attachments": {
        "SOLR-6868.patch": "https://issues.apache.org/jira/secure/attachment/12688776/SOLR-6868.patch"
    },
    "issue_links": {},
    "comments": [
        {
            "date": "2014-12-23T02:48:42+0000",
            "author": "Anshum Gupta",
            "content": "For one of the failures, seems like the collection created for one of the test methods was interfering with the working of the rest.\nThis should fix the assertionFailure that we've also seen on Jenkins every once in a while. ",
            "id": "comment-14256506"
        },
        {
            "date": "2014-12-23T02:52:08+0000",
            "author": "ASF subversion and git services",
            "content": "Commit 1647472 from Anshum Gupta in branch 'dev/trunk'\n[ https://svn.apache.org/r1647472 ]\n\nSOLR-6868: Try to fix the failing test and also a part of the SolrJ collection API test refactoring ",
            "id": "comment-14256517"
        },
        {
            "date": "2014-12-23T02:54:02+0000",
            "author": "ASF subversion and git services",
            "content": "Commit 1647474 from Anshum Gupta in branch 'dev/branches/branch_5x'\n[ https://svn.apache.org/r1647474 ]\n\nSOLR-6868: Try to fix the failing test and also a part of the SolrJ collection API test refactoring (merge from trunk) ",
            "id": "comment-14256522"
        },
        {
            "date": "2015-12-06T14:10:56+0000",
            "author": "Shalin Shekhar Mangar",
            "content": "Looks like this particular failure was fixed. Can you close this issue? ",
            "id": "comment-15043904"
        },
        {
            "date": "2015-12-07T07:05:53+0000",
            "author": "Anshum Gupta",
            "content": "Done. ",
            "id": "comment-15044464"
        }
    ]
}